experimental/Geometry/Multiindices.jl

```diff
@@ -0,0 +1,221 @@
+module Multiindices
+
+export OrderedMultiindex
+export linear_index
+export index_signature
+export HomogMultiindex
+export power
+export getindex
+
+
+abstract type Multiindex end
+
+mutable struct OrderedMultiindex <: Multiindex 
+  n::Int
+  p::Int
+  index::Vector{Int}
+  function OrderedMultiindex(n::Int, p::Int)
+    v = Vector{Int}(undef,p) 
+    for i in 1:p
+      v[i] = i
+    end
+    #@show n, p, v
+    return new( n, p, v )
+  end
+end
+
+function Base.show( io::Base.TTY, i::OrderedMultiindex ) 
+  outstr= "0"
+  for k in 1:length(i.index) 
+    outstr = outstr * " < " * string(i.index[k]) 
+  end
+  outstr = outstr * " <= " * string( i.n ) * "\n"
+  #Base.show( io, "n = " * string(i.n) * "; p = " * string(i.p) * "; index = " * string( i.index ) )
+  Base.print( io, outstr )
+end
+
+function Base.iterate( i::OrderedMultiindex ) 
+  for k in 1:i.p
+    i.index[k] = k
+  end
+  return i, i.index
+end
+
+function Base.iterate( i::OrderedMultiindex, state ) 
```

Comment:
I think using 'state' is optional.

Reply:
No

Reply:
Is the `OrderedMultiindex` supposed to represent on index `a1 < ... < ak` or the set of all of these indices? I am confused because the object itself is returned from the iterator.

Reply:
Both (see above): It has the datum for the range stored in n and p and then the datum for the actual state in i (resp. a). 

Keeping the knowledge about n and p is important in order to make sense of the values for a and, for instance, to calculate the associated linear index, etc. I'll try to write some more explanatory comments in the source file and then hopefully things will be more clear.